rust/xaynet-analytics/src/database/analytics_event/repo.rs

```diff
@@ -0,0 +1,51 @@
+use anyhow::{Error, Result};
+
+use crate::database::{
+    analytics_event::data_model::AnalyticsEvent,
+    common::{IsarAdapter, Repo},
+    isar::IsarDb,
+};
+
+pub struct AnalyticsEventRepo {
+    collection_name: String,
+    db: IsarDb,
+}
+
+impl AnalyticsEventRepo {
+    pub fn new(collection_name: String, db: IsarDb) -> Self {
+        Self {
+            collection_name,
+            db,
+        }
+    }
+}
+
+impl Repo<AnalyticsEvent> for AnalyticsEventRepo {
+    fn add(&self, event: AnalyticsEvent) -> Result<(), Error> {
+        let mut object_builder = self.db.get_object_builder(&self.collection_name)?;
+        event.write_with_object_builder(&mut object_builder);
+        self.db
+            .put(&self.collection_name, object_builder.finish().as_bytes())
+            .map(|_| ())
+    }
+
+    // TODO: return an iterator instead of Vec: https://xainag.atlassian.net/browse/XN-1517
+    fn get_all(&self) -> Result<Vec<AnalyticsEvent>, Error> {
+        let _events_as_bytes = self.db.get_all_as_bytes(&self.collection_name)?;
+
+        // TODO: not sure how to proceed to parse [u8] using the collection schema. didn't find examples in Isar
```

Comment:
Do we have ticket for it?

Reply:
nup, i'll add it.

https://xainag.atlassian.net/browse/XN-1604

i'll add it in the code too